src/client/opamClient.ml

```diff
@@ -752,6 +752,62 @@ let init_checks ?(hard_fail_exn=true) init_config =
   if hard_fail && hard_fail_exn then OpamStd.Sys.exit_because `Configuration_error
   else not (soft_fail || hard_fail)
 
+(* Check sandboxing script call. If it errors or unattended output, disable
+   sandboxing by removing [OpamInitDefaults.sandbox_wrappers] commands in
+   config file.
+   Only one script is checked (init script default one), and tested on an
+   `echo SUCCESS' call.
+*)
+let check_and_revert_sandboxing root config =
+  let sdbx_wrappers =
+    let w = OpamFile.Config.wrappers config in
+    let init_sdbx_cmds =
+      List.map (function `build cmd | `install cmd | `remove cmd -> cmd)
+        OpamInitDefaults.sandbox_wrappers
+      |> List.flatten
+    in
+    List.filter (fun cmd -> List.mem cmd init_sdbx_cmds)
+      OpamFile.Wrappers.(wrap_build w @ wrap_install w @ wrap_remove w)
+  in
+  let hook_var = OpamVariable.of_string "hooks" in
+  let hook_val = Some (OpamVariable.dirname (OpamPath.hooks_dir root)) in
+  let env = fun v ->
+    let fv = OpamVariable.Full.variable v in
+    if fv = hook_var then hook_val else
+      OpamStd.Option.Op.(OpamStd.Option.of_Not_found (List.assoc fv)
+                           OpamSysPoll.variables >>= Lazy.force)
+  in
+  match OpamFilter.commands env sdbx_wrappers with
+  | [] -> config
```

Comment:
Too bad we can't use the code of `OpamAction` here :/

Reply:
Added `OpamEnv.hook_env`